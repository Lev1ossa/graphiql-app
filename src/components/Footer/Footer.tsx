import { userAvatars } from '@/constants/constants'
import { Avatar, Badge, Link, Paper, Stack, styled } from '@mui/material'
import { RSSLogo } from './RSSLogo'

const StyledBadge = styled(Badge)<{ active?: boolean }>(
  ({ theme, active = false }) => ({
    overflow: 'hidden',
    ...(active && {
      '& .MuiBadge-badge': {
        color: theme.palette.success,
        backgroundColor: theme.palette.success.main,
        boxShadow: `0 0 0 3px ${theme.palette.background.paper}`,
      },
    }),
  }),
)

export function Footer() {
  return (
    <Paper>
      <Stack
        direction="row"
        py={0.5}
        px={2}
        spacing={1}
        justifyContent="space-between"
      >
        <Stack direction="row" spacing={1} alignItems="center">
          {userAvatars.map(
            ({ href, imgSrc, imgAlt, active = false }, index) => (
              <Link key={index} href={href}>
                <StyledBadge
                  active={active}
                  overlap="circular"
                  anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
                  variant="dot"
                >
                  <Avatar variant="rounded" src={imgSrc} alt={imgAlt} />
                </StyledBadge>
              </Link>
            ),
          )}
        </Stack>
        <RSSLogo />
      </Stack>
    </Paper>
  )
}
