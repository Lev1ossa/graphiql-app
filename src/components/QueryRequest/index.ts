export { default as QueryRequest } from './QueryRequest'
