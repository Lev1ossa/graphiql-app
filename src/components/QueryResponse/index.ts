export { default as QueryResponse } from './QueryResponse'
