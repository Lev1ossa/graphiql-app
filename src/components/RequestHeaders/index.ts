export { default as RequestHeaders } from './RequestHeaders'
