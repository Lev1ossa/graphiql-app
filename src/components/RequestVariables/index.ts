export { default as RequestVariables } from './RequestVariables'
