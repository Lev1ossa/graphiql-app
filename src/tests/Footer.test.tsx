import { Footer } from '@/components/Footer'
import { expect, test } from 'vitest'
import renderInContext from './renderInContext'

test('Correct Footer snapshot', () => {
  const { baseElement } = renderInContext(<Footer />)
  expect(baseElement).toMatchSnapshot()
})
