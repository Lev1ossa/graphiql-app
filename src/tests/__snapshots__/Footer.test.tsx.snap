// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Correct Footer snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
    >
      <div
        class="MuiStack-root css-1eoeowx-MuiStack-root"
      >
        <div
          class="MuiStack-root css-kdc3n8-MuiStack-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="https://github.com/Alyona8891"
          >
            <span
              class="MuiBadge-root css-1arcc7-MuiBadge-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-rounded css-12bapnq-MuiAvatar-root"
              >
                <img
                  alt="Alyona's avatar"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="https://avatars.githubusercontent.com/u/79936182?v=4"
                />
              </div>
              <span
                class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-s1rlol-MuiBadge-badge"
              />
            </span>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="https://github.com/Lev1ossa"
          >
            <span
              class="MuiBadge-root css-1arcc7-MuiBadge-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-rounded css-12bapnq-MuiAvatar-root"
              >
                <img
                  alt="Alexandr's avatar"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="https://avatars.githubusercontent.com/u/104268412?v=4"
                />
              </div>
              <span
                class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-s1rlol-MuiBadge-badge"
              />
            </span>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="https://github.com/lazy-goose"
          >
            <span
              class="MuiBadge-root css-3jxqn3-MuiBadge-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-rounded css-12bapnq-MuiAvatar-root"
              >
                <img
                  alt="Maksim's avatar"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="https://avatars.githubusercontent.com/u/119844669?v=4"
                />
              </div>
              <span
                class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-s1rlol-MuiBadge-badge"
              />
            </span>
          </a>
        </div>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-uwx3rb-MuiTypography-root-MuiLink-root"
          href="https://rs.school/react"
        >
          <time
            class="MuiBox-root css-1h14jth"
          >
            2024
          </time>
          <div
            class="MuiBox-root css-psrz38"
          >
            <span>
              RS
            </span>
            <span>
              SC HO OL
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</body>
`;
